packages/shared/src/hooks/notifications/useBookmarkReminder.ts

```diff
@@ -0,0 +1,104 @@
+import { useMutation } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { addDays, addHours } from 'date-fns';
+import {
+  setBookmarkReminder,
+  SetBookmarkReminderProps,
+} from '../../graphql/bookmarks';
+import { useToastNotification } from '../useToastNotification';
+
+export enum BookmarkReminderPreference {
+  OneHour = 'In 1 hour',
+  LaterToday = 'Later today',
+  Tomorrow = 'Tomorrow',
+  TwoDays = 'In 2 days',
+  NextWeek = 'Next week',
+}
+
+interface MutateBookmarkProps extends SetBookmarkReminderProps {
+  existingReminder?: Date;
+  preference: BookmarkReminderPreference;
+}
+
+interface BookmarkReminderProps {
+  postId: string;
+  preference: BookmarkReminderPreference;
+  existingReminder?: Date;
+}
+
+interface UseBookmarkReminder {
+  onBookmarkReminder: (props: BookmarkReminderProps) => void;
+}
+
+export function getNextMonday(date: Date): Date {
+  const nextMonday = new Date(date);
+  const day = date.getDay();
+  const daysUntilNextMonday = (8 - day) % 7 || 7; // 8 % 7 = 1 (if today is Monday, get the next Monday)
```

Comment:
I think you can use `nextDay` from `date-fns`, should be easier.

Reply:
Oh, let me check their docs about it.